addons/account/models/account_move_line.py

```diff
@@ -1353,7 +1353,16 @@ def _sanitize_vals(self, vals):
                 vals.pop('debit', None)
                 vals.pop('credit', None)
             else:
-                vals['balance'] = vals.pop('debit', 0) - vals.pop('credit', 0)
+                move = self.env['account.move']
+                if 'move_id' in vals:
+                    move = self.env['account.move'].browse(vals['move_id'])
```

Comment:
I'm wondering if this is working while we're creating the move and its lines.
If there isn't a `move_id`, maybe there is a `move_type` and a `currency_id` in it that could help us to still retrieve the right data to make the conversion.

Reply:
Agreed, I pushed an attempt to retrieve the needed fields, let me know what you think about it